tests/ruling/utils.ts

```diff
@@ -0,0 +1,194 @@
+  const expectedRules = Object.keys(expected);
+  const actual: Results = readSnapshots(expectedRules);
+  const actualRules = Object.keys(actual);
+  let passed = true;
+
+  expectedRules.forEach(rule => {
+    const expectedFiles = expected[rule];
+    const actualFiles = actual[rule] || {};
+
+    Object.keys(expectedFiles).forEach(file => {
+      const expectedLines = expectedFiles[file];
+      const actualLines = actualFiles[file] || [];
+
+      const missingLines = lodash.difference(actualLines, expectedLines);
+      if (missingLines.length > 0) {
+        passed = false;
+        console.log("Missing issues:");
+        console.log("  * Rule:", rule);
+        console.log("  * File:", file);
+        console.log("  * Lines:", missingLines.join(", "));
+        console.log();
+      }
+
+      const extraLines = lodash.difference(expectedLines, actualLines);
+      if (extraLines.length > 0) {
+        passed = false;
+        console.log("Extra issues:");
+        console.log("  * Rule:", rule);
+        console.log("  * File:", file);
+        console.log("  * Lines:", extraLines.join(", "));
+        console.log();
+      }
+    });
+  });
+
+  return passed;
+}
+
+function initRule(Rule: any): tslint.IRule {
+  const RULE_OPTIONS: tslint.IOptions = {
+    disabledIntervals: [],
+    ruleArguments: [],
+    ruleName: "",
+    ruleSeverity: "error",
+  };
+  return new Rule(RULE_OPTIONS);
+}
+
+function getProgram(tsConfigFile: string): ts.Program {
+  return tslint.Linter.createProgram(tsConfigFile);
+}
+
+function getProgramFiles(program: ts.Program): ts.SourceFile[] {
+  return program.getSourceFiles().filter(file => !file.isDeclarationFile);
+}
+
+function runRuleOnProjectFile(rule: any, sourceFile: ts.SourceFile, program: ts.Program) {
+  let failures: tslint.RuleFailure[];
+
+  if ((rule as tslint.Rules.TypedRule).applyWithProgram) {
+    failures = rule.applyWithProgram(sourceFile, program);
+
+  } else {
+    failures = rule.apply(sourceFile);
+  }
+
+  return failures.map(failure => failure.getStartPosition().getLineAndCharacter().line + 1);
+}
+
+function mergeErrorsToResults(
+  results: Results,
+  ruleName: string,
+  fileName: string,
+  errorLines: number[],
+): Results {
+  if (errorLines.length > 0) {
+    const nextResults = { ...results };
+
+    if (!(ruleName in nextResults)) {
+      nextResults[ruleName] = {};
+    }
+
+    if (!(fileName in nextResults[ruleName])) {
+      nextResults[ruleName][fileName] = [];
+    }
+
+    nextResults[ruleName][fileName] = [
+      ...nextResults[ruleName][fileName],
+      ...errorLines,
+    ];
+
+    return nextResults;
+  } else {
+    return results;
+  }
+}
+
+function getFileNameForSnapshot(path: string): string {
+  const marker = "/typescript-test-sources/";
+  const pos = path.indexOf(marker);
+  return path.substr(pos + marker.length);
+}
+
+function writeSnapshot(rule: string, content: string): void {
+  const fileName = path.join(__dirname, "snapshots", "rule");
```

Comment:
Remove the quotes around `"rule"`. Right now we are writing only to file `"rule"` :)